gradio/chat_interface.py

```diff
@@ -0,0 +1,268 @@
+            self._setup_api()
+
+    def _setup_events(self):
+        if inspect.isgeneratorfunction(self.fn):
+            submit_fn = self._stream_fn
+        else:
+            submit_fn = self._submit_fn
+
+        self.textbox.submit(
+            self._clear_and_save_textbox,
+            [self.textbox],
+            [self.textbox, self.saved_input],
+            api_name=False,
+            queue=False,
+        ).then(
+            self._display_input,
+            [self.saved_input, self.chatbot],
+            [self.chatbot],
+            api_name=False,
+            queue=False,
+        ).then(
+            submit_fn,
+            [self.saved_input, self.chatbot],
+            [self.chatbot],
+            api_name=False,
+        )
+
+        if self.submit_btn:
+            self.submit_btn.click(
+                self._clear_and_save_textbox,
+                [self.textbox],
+                [self.textbox, self.saved_input],
+                api_name=False,
+                queue=False,
+            ).then(
+                self._display_input,
+                [self.saved_input, self.chatbot],
+                [self.chatbot],
+                api_name=False,
+                queue=False,
+            ).then(
+                submit_fn,
+                [self.saved_input, self.chatbot],
+                [self.chatbot],
+                api_name=False,
+            )
+        
+        if self.retry_btn:
+            self.retry_btn.click(
+                self._delete_prev_fn,
+                [self.chatbot],
+                [self.chatbot, self.saved_input],
+                api_name=False,
+                queue=False,
+            ).then(
+                self._display_input,
+                [self.saved_input, self.chatbot],
+                [self.chatbot],
+                api_name=False,
+                queue=False,
+            ).then(
+                submit_fn,
+                [self.saved_input, self.chatbot],
+                [self.chatbot],
+                api_name=False,
+            )
+
+        if self.delete_last_btn:
+            self.delete_last_btn.click(
+                self._delete_prev_fn,
+                [self.chatbot],
+                [self.chatbot, self.saved_input],
+                api_name=False,
+                queue=False,
+            )
+
+        if self.clear_btn:
+            self.clear_btn.click(
+                lambda :([], None), 
+                None, 
+                [self.chatbot, self.saved_input],
+                queue=False,
+                api_name=False)
+            
+    def _setup_api(self):
+        if inspect.isgeneratorfunction(self.fn):
+            api_fn = self._api_stream_fn
+        else:
+            api_fn = self._api_submit_fn
+
+        # Use a gr.State() instead of self.chatbot so that the API doesn't require passing forth
+        # a chat history, instead it is just stored internally in the state.
+        history = State([])
+
+        self.fake_api_btn.click(
```

Comment:
Something's not right here - the view api page and client both show that this route takes and returns nothing

<img width="405" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/9e25ee10-5978-4b48-a864-d7c0537506b4">

<img width="943" alt="image" src="https://github.com/gradio-app/gradio/assets/41651716/9d4e733d-35de-4c7b-84cb-1effaa92dd2c">

This is with this demo

```python
from gradio import ChatInterface
import time

def double(message, history):
    time.sleep(1)
    return message + message

ChatInterface(double, examples=["bbbb", "abcdef"], cache_examples=True).queue().launch()
```

Reply:
Should be fixed now, thanks @freddyaboulton!